rclcpp/include/rclcpp/mapped_ring_buffer.hpp

```diff
@@ -101,53 +102,91 @@ class MappedRingBuffer : public MappedRingBufferBase
    * \param value if the key is found, the value is stored in this parameter
    */
   void
-  get_copy_at_key(uint64_t key, ElemUniquePtr & value)
+  get(uint64_t key, ElemUniquePtr & value)
```

Comment:
Since you changed the method name here, is this actually changing the behavior?

We need to update the documentation here to describe the new behavior if so.

Reply:
This is the renaming I did:
get_copy_at_key -> get
pop_at_key -> pop
get_ownership_at_key -> Deleted. It was used nowhere.

get, depending on the signature used and the stored message, could return a copy or share the ownership.
pop gives the ownership of the message to the caller. Except when storing a shared_ptr and trying to pop it into unique_ptr, as it's impossible to downgrade a shared_ptr. In this case, the shared_ptr is copied into the unique_ptr, and the MappedRingBuffer releases its ownership of the shared_ptr.

I will update the documentation, because I see it's wrong in some places.